src/irr/asset/CGLTFLoader.cpp

```diff
@@ -29,39 +37,53 @@ namespace irr
 			_IRR_STATIC_INLINE_CONSTEXPR uint8_t MAX_WEIGHTS_ATTRIBUTES = 4;
 		}
 
-		namespace SShaderPaths
-		{
-			namespace UV
-			{
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view vertex = ""; // TODO - include/builtin/...
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view fragment = ""; // TODO - include/builtin/...
-			}
-
-			namespace Color
-			{
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view vertex = ""; // TODO - include/builtin/...
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view fragment = ""; // TODO - include/builtin/...
-			}
-
-			namespace NoUVAndColor
-			{
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view vertex = ""; // TODO - include/builtin/...
-				_IRR_STATIC_INLINE_CONSTEXPR std::string_view fragment = ""; // TODO - include/builtin/...
-			}
-		}
-
 		/*
 			Each glTF asset must have an asset property. 
 			In fact, it's the only required top-level property
 			for JSON to be a valid glTF.
 		*/
 
+		CGLTFLoader::CGLTFLoader(asset::IAssetManager* _m_assetMgr) 
+			: assetManager(_m_assetMgr)
+		{
+			auto registerShader = [&](auto constexprStringType, ICPUSpecializedShader::E_SHADER_STAGE stage) -> void
+			{
+				auto shaderData = assetManager->getFileSystem()->loadBuiltinData<decltype(constexprStringType)>();
+				auto unspecializedShader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(shaderData), asset::ICPUShader::buffer_contains_glsl);
+
+				ICPUSpecializedShader::SInfo specInfo({}, nullptr, "main", stage, stage != ICPUSpecializedShader::ESS_VERTEX ? "?IrrlichtBAW glTFLoader FragmentShader?" : "?IrrlichtBAW glTFLoader VertexShader?");
+				auto cpuShader = core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(unspecializedShader), std::move(specInfo));
+
+				auto insertShaderIntoCache = [&](const char* path)
+				{
+					asset::SAssetBundle bundle({ cpuShader });
+					assetManager->changeAssetKey(bundle, path);
+					assetManager->insertAssetIntoCache(bundle);
```

Comment:
builtin's like MTL now will use a different asset insertion mechanism (not ready yet), subscribe to https://github.com/buildaworldnet/IrrlichtBAW/pull/549

Reply:
the API is ready